### DISCUSSION TITLE
Some parameters are not working on WebP outputs

### DISCUSSION THREAD

**Discussion by quentin-bettoum** [libvips/libvips#3341] [General]
It seems like some parameters such as `strip`, `min-size`, or `mixed` don't have any effect on the output file for the **WebP** format.

For exemple, with the `strip` parameter, weither I add it or not, the **WebP** output will keep all metadata from the original image.

An exemple using the cli, with or without `strip`, the amount of exif data on the output .webp is the same unlike other formats.
```bash
vips thumbnail forest.jpg forest-stripped.webp[Q=50,strip] 500
vips thumbnail forest.jpg forest-not-stripped.webp[Q=50] 500
exiftool forest-stripped.webp | wc -l
exiftool forest-not-stripped.webp | wc -l
```

If I use the same command with another output format (in this case `AVIF`), the strip argument does remove a lot of metadata which is the expected behaviour.
```bash
vips thumbnail forest.jpg forest-stripped.avif[Q=50,strip] 500
vips thumbnail forest.jpg forest-not-stripped.avif[Q=50] 500
exiftool forest-stripped.avif | wc -l
exiftool forest-not-stripped.avif | wc -l
```

**Comment by lovell** [comment]
Please can you provide a sample image and details of all version numbers (libvips, libwebp etc.)

**Comment by quentin-bettoum** [comment]
Sure, I am using `libvips 8.14.1-1` and `libwebp 1.3.0-2`.

Here is a sample image
![forest](https://user-images.githubusercontent.com/63603941/219120701-0412baa6-ce31-4600-a373-8748abf7c689.jpg)

**Comment by jcupitt** [comment]
Ah the strip parameter wasn't connected up correctly, looks like the last refactoring broke it silently.

Should be fixed with: https://github.com/libvips/libvips/commit/6715108a2dc45c04634ab0ed545ccac6a63e7592

```
john@banana ~/pics $ vips copy k2.jpg x.webp
john@banana ~/pics $ vips copy k2.jpg x2.webp[strip]
john@banana ~/pics $ ls -l x*.webp
-rw-r--r-- 1 john john 12116228 Feb 15 19:20 x2.webp
-rw-r--r-- 1 john john 12119592 Feb 15 19:19 x.webp
```

**Reply by lovell** [reply]
Thanks John, should we backport this to the 8.14 branch?

**Reply by kleisauke** [reply]
Backport (+ changelog entry) to [`8.14`](https://github.com/libvips/libvips/tree/8.14) available as #3343.

**Reply by quentin-bettoum** [reply]
Thanks, and as for `min-size` and `mixed`, by reading the code I'm guessing these parameters should only be used on animated webp?
That would explain why I didn't see any effect in my case.

**Reply by kleisauke** [reply]
Indeed, the `min-size` and `mixed` parameters only applies to animated WebP output.
https://github.com/libvips/libvips/blob/02213d4031ac580d4fdb3ccce1306155657aeedf/libvips/foreign/webpsave.c#L1129-L1137
